Development/nmos/ocsp_behaviour.cpp

```diff
@@ -0,0 +1,366 @@
+        {
+            slog::log<slog::severities::too_much_info>(gate, SLOG_FLF) << "Requesting certificate status";
+
+            using namespace web::http;
+
+            http_request req(methods::POST);
+            req.headers().add(header_names::content_type, U("application/ocsp-request"));
+            req.set_body(ocsp_request);
+
+            return nmos::api_request(client, req, gate, token).then([=, &gate](pplx::task<http_response> response_task)
+            {
+                auto response = response_task.get(); // may throw http_exception
+
+                if (status_codes::OK == response.status_code())
+                {
+                    if (response.body())
+                    {
+                        return response.extract_vector().then([&gate](std::vector<uint8_t> body)
+                        {
+                            slog::log<slog::severities::too_much_info>(gate, SLOG_FLF) << "Received OCSP response";
+                            return body;
+
+                        }, token);
+                    }
+                    else
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "OCSP certificate status request error: missing OCSP response";
+                        throw ocsp_service_exception();
+                    }
+                }
+                else
+                {
+                    slog::log<slog::severities::error>(gate, SLOG_FLF) << "OCSP certificate status request error: " << response.status_code() << " " << response.reason_phrase();
+                    throw ocsp_service_exception();
+                }
+            }, token);
+        }
+
+        // task to continuously fetch the certificate status (OCSP response) on a time interval until failure or cancellation
+        pplx::task<void> do_certificate_status_requests(nmos::model& model, nmos::experimental::ocsp_state& ocsp_state, ocsp_shared_state& state, slog::base_gate& gate, const pplx::cancellation_token& token = pplx::cancellation_token::none())
+        {
+            const auto& ocsp_interval_min(nmos::experimental::fields::ocsp_interval_min(model.settings));
+            const auto& ocsp_interval_max(nmos::experimental::fields::ocsp_interval_max(model.settings));
+
+            // start a background task to continously request certificate status on a given interval
+            return pplx::do_while([=, &model, &ocsp_state, &state, &gate]
+            {
+                auto request_interval = std::chrono::seconds(0);
+                if (state.base_uri == state.client->base_uri())
+                {
+                    auto interval = std::uniform_int_distribution<>(
+                        std::min(ocsp_interval_min, (state.next_request > 0.0 ? (int)state.next_request : ocsp_interval_min)),
+                        std::min(ocsp_interval_max, (state.next_request > 0.0 ? (int)state.next_request : ocsp_interval_max)))(state.engine);
```

Comment:
I find this slightly easier to read. What do you think?

```suggestion
                        state.next_request > 0.0 ? std::min((int)state.next_request, ocsp_interval_min) : ocsp_interval_min,
                        state.next_request > 0.0 ? std::min((int)state.next_request, ocsp_interval_max) : ocsp_interval_max)(state.engine);
```